src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_SIZE = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V>[] thisTable;
+        Node<K, V> tableNode;
+        int tableLength;
+        if ((thisTable = table) == null || (tableLength = thisTable.length) == 0) {
+            tableLength = (thisTable = resize()).length;
+        }
+        int index = indexFor(hash(key), tableLength);
+        if ((tableNode = thisTable[index]) == null) {
+            thisTable[index] = newNode(hash(key), key, value);
+        } else {
+            Node<K, V> node;
+            if (tableNode.key != null && tableNode.key.equals(key) || tableNode.key == key) {
+                node = tableNode;
+            } else {
+                for ( ; ; ) {
+                    if ((node = tableNode.next) == null) {
+                        tableNode.next = newNode(hash(key), key, value);
+                        break;
+                    }
+                    if (node.key != null && node.key.equals(key) || node.key == key) {
+                        break;
+                    }
+                    tableNode = node;
+                }
+            }
+            if (node != null) {
+                node.value = value;
+                return;
+            }
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V>[] thisTable;
+        if ((thisTable = table) == null || thisTable.length == 0) {
+            return null;
+        }
+        int index = indexFor(hash(key),thisTable.length);
```

Comment:
you don't need additional checks. If table[index] is null, the loop never executes and you get null from the method.  
```suggestion
        int index = indexFor(hash(key),thisTable.length);
        Node<K, V>[] thisTable = table[index];
```